astroid/nodes/node_ng.py

```diff
@@ -248,15 +262,34 @@ def parent_of(self, node):
                 return True
         return False
 
-    def statement(self):
+    @overload
+    def statement(
+        self, future: Literal[False] = ...
+    ) -> Union["nodes.Statement", "nodes.Module"]:
+        ...
```

Comment:
I believe we should add an additional overload before that.
```py
    @overload
     def statement(self) -> Union["nodes.Statement", "nodes.Module"]:
         ...
```
This should match those cases were `future` isn't passed.

Reply:
I think the `...` after `Literal[False]` does this? I played around with it and this does seem to work. VS Code picks up on it and `mypy` does as well I think.

The screenshot is from `nodes.Module`, but I think it should work similarly.
<img width="734" alt="Schermafbeelding 2021-11-03 om 16 35 52" src="https://user-images.githubusercontent.com/13665637/140092477-d38b5a5b-e635-4eab-9b9e-aba1036eccf5.png">

Reply:
Tested a few things. What do you think about this solution?
```py
    @overload
    def statement(
        self, future: Literal[None] = ...
    ) -> Union["nodes.Statement", "nodes.Module"]:
        ...

    @overload
    def statement(self, future: Literal[True]) -> "nodes.Statement":
        ...

    def statement(
        self, future: Optional[Literal[True]] = None
    ) -> Union["nodes.Statement", "nodes.Module"]:
```

And this for `Module`:
```py
    @overload
    def statement(self, future: Literal[None] = ...) -> "Module":
        ...

    @overload
    def statement(self, future: Literal[True]) -> NoReturn:
        ...

    def statement(self, future: Literal[True] | None = None) -> Union[NoReturn, "Module"]:
```

Reply:
Note that I also changed the default to `None`.

Reply:
Looks good to me! I'll create a commit.

Reply:
Did you test these with `pylint`?
I'm getting the following error with pre-commit:

```shell
************* Module astroid.nodes.scoped_nodes
astroid/nodes/scoped_nodes.py:659:4: W0222: Signature differs from overridden 'statement' method (signature-differs)
```

Reply:
I did not. It's not immediately obvious to me what's causing the issue. Need to investigate that a bit more.